fvm/fvm_test.go

```diff
@@ -1825,6 +1825,331 @@ func TestSignatureVerification(t *testing.T) {
+						case crypto.ECDSAP256, crypto.ECDSASecp256k1:
+							assert.NoError(t, err)
+							assert.Error(t, script.Err)
+						}
+					})
+					t.Run("incorrect key", func(t *testing.T) {
+						pk, err := crypto.GeneratePrivateKey(signatureAlgorithm.algorithm, seed)
+						require.NoError(t, err)
+
+						publicKey := testutil.BytesToCadenceArray(
+							pk.PublicKey().Encode(),
+						)
+						privateKey := testutil.BytesToCadenceArray(
+							pk.Encode(),
+						)
+						script := fvm.Script(code).WithArguments(
+							jsoncdc.MustEncode(publicKey),
+							jsoncdc.MustEncode(privateKey),
+						)
+
+						err = vm.Run(ctx, script, view, programs)
+						switch signatureAlgorithm.algorithm {
+						case crypto.BLSBLS12381:
+							assert.NoError(t, err)
+							assert.NoError(t, script.Err)
+							assert.Equal(t, cadence.NewBool(false), script.Value)
+						case crypto.ECDSAP256, crypto.ECDSASecp256k1:
+							assert.NoError(t, err)
+							assert.Error(t, script.Err)
+						}
+					})
+				},
+			))
+		})
+		t.Run(signatureAlgorithm.name, func(t *testing.T) {
+			t.Run("aggregateBLSSignatures", newVMTest().run(
+				func(
+					t *testing.T,
+					vm *fvm.VirtualMachine,
+					chain flow.Chain,
+					ctx fvm.Context,
+					view state.View,
+					programs *programs.Programs,
+				) {
+
+					code := []byte(
+						`
+							import Crypto
+	
+							pub fun main(
+							signatures: [[UInt8]],
+							): [UInt8] {
+								return AggregateBLSSignatures(signatures)
+							}
+						`,
+					)
+					t.Run("same message", func(t *testing.T) {
+						// random message
+						input := make([]byte, 100)
+						_, err := rand.Read(input)
+						require.NoError(t, err)
+
+						if hashAlgorithm.name != "KMAC128_BLS_BLS12_381" {
+							input = append(
+								flow.UserDomainTag[:],
+								input...,
+							)
+						}
+
+						numSigs := 100
+
+						signatures := make([]cadence.Value, 0, numSigs)
+						sks := make([]crypto.PrivateKey, 0, numSigs)
+
+						for i := 0; i < numSigs; i++ {
+							sk := randomSK(t, seed)
+							s, err := sk.Sign(input, hasher)
+							require.NoError(t, err)
+
+							sks = append(sks, sk)
+							sig := testutil.BytesToCadenceArray(s)
+							signatures = append(signatures, sig)
+						}
+
+						script := fvm.Script(code).WithArguments(
+							jsoncdc.MustEncode(cadence.Array{
+								Values: signatures,
+								ArrayType: cadence.VariableSizedArrayType{
+									ElementType: cadence.VariableSizedArrayType{
+										ElementType: cadence.UInt8Type{},
+									},
+								},
+							}),
+						)
+
+						err = vm.Run(ctx, script, view, programs)
+						switch signatureAlgorithm.algorithm {
+						case crypto.BLSBLS12381:
+							assert.NoError(t, err)
+							assert.NoError(t, script.Err)
+
+							aggSk, err := crypto.AggregateBLSPrivateKeys(sks)
+							require.NoError(t, err)
+							expectedSig, err := aggSk.Sign(input, hasher)
+							require.NoError(t, err)
```

Comment:
Nice solution! 
Another (faster) way would be to aggregate the signatures straight away.

```suggestion
							expectedSig, err := crypto.AggregateBLSSignatures(sigs)
							require.NoError(t, err)
```

Reply:
Well, I figured we will need to construct `sigs` too.